src/System.Text.Primitives/System/Text/InvariantParser_bool.cs

```diff
@@ -83,6 +156,91 @@ private static bool IsFalse(byte[] utf8Text, int index)
 
             return true;
         }
+
+        private static bool IsFalse(string text, int index)
+        {
+            if (text.Length - index < 5)
+                return false;
+
+            char firstChar = text[index];
+            if (firstChar != 'f' && firstChar != 'F')
+                return false;
+
+            char secondChar = text[index + 1];
+            if (secondChar != 'a' && secondChar != 'A')
+                return false;
+
+            char thirdChar = text[index + 2];
+            if (thirdChar != 'l' && thirdChar != 'L')
+                return false;
+
+            char fourthChar = text[index + 3];
+            if (fourthChar != 's' && fourthChar != 'S')
+                return false;
+
+            char fifthChar = text[index + 4];
+            if (fifthChar != 'e' && fifthChar != 'E')
+                return false;
+
+            return true;
+        }
+
+        private static bool IsFalse(Utf8String text)
+        {
+            if (text.Length < 5)
+                return false;
+
+            Utf8CodeUnit firstChar = text[0];
+            if (firstChar != 'f' && firstChar != 'F')
+                return false;
+
+            Utf8CodeUnit secondChar = text[1];
+            if (secondChar != 'a' && secondChar != 'A')
+                return false;
+
+            Utf8CodeUnit thirdChar = text[2];
+            if (thirdChar != 'l' && thirdChar != 'L')
+                return false;
+
+            Utf8CodeUnit fourthChar = text[3];
+            if (fourthChar != 's' && fourthChar != 'S')
+                return false;
+
+            Utf8CodeUnit fifthChar = text[4];
+            if (fifthChar != 'e' && fifthChar != 'E')
+                return false;
+
+            return true;
+        }
+
+        private static bool IsFalse(ReadOnlySpan<byte> utf8Bytes)
```

Comment:
If we're going to have a lot of overloads of these helper methods (w/ essentially identical code except for types), we might look into putting these in a T4 template so that the code is more maintainable.